sphericart/src/sphericart_cuda.cpp

```diff
@@ -0,0 +1,188 @@
+#include <iostream>
+#include <stdexcept>
+
+#define _SPHERICART_INTERNAL_IMPLEMENTATION
+
+#include "dynamic_cuda.hpp"
+#include "sphericart.hpp"
+#include "cuda_base.hpp"
+#include "sphericart_cuda.hpp"
+
+using namespace sphericart::cuda;
+
+/*
+    This code checks whether the cuda libraries we'd need to dynamically load are available on the host
+*/
+
+void checkCuda() {
+    if (!cudadriver.loaded()) {
+        throw std::runtime_error(
+            "Failed to load libcuda.so. Try running \"find /usr -name libcuda.so\" and "
+            "appending the directory to your $LD_LIBRARY_PATH environment variable."
+        );
```

Comment:
This is duplicated from the `CUDA` class constructor, I would prefer if the same error message did not exist twice in the same code base.

Reply:
this is resolved now, only sphericart_cuda.cpp will throw this error.